### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?

In this PR I create a Completed Stock Screener Website which is a tool quickly find current stock prices in USA companies...

Description of the task to be completed 👎

**Setup the project with React.**
**Setup Redux Store**
**Fetch the API to the Store**
**Dispatch the API to the Components**
**Apply the Desing**

### CODE REVIEW

File: src/pages/Home.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const handleSelection = (company) => {
+    const filtered = filterCompany(company, stockMaterials);
+    setStockMaterial(filtered);
+  };
+
+  const handleSearch = (text) => {
+    // eslint-disable-next-line
+    const filteredCompanies = stockMaterials.filter((ex) => ex.companyName.toLowerCase().includes(text));
+    setStockMaterial(filteredCompanies);
+  };
```
Comment (amejid) [inline src/pages/Home.js]: - [x] Great job with the design and the functionalities. :+1: There seems to be a bug in the search functionality. The moment I start writing in the text field, every card is gone. Kindly fix it. 🧐️
- [x] Please remove the line where ESLint is disabled as they help us in achieving a cleaner code. 🤝️

File: src/_tests_/Home.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const tree = renderer
+    .create(
+      <Provider store={store}>
+        <Router>
+          <Home />
+        </Router>
+      </Provider>,
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
```
Comment (amejid) [inline src/_tests_/Home.test.js:21]: - [x] One of the tests is failing, let's make sure all the tests are passing. ✍️

Screenshot from 2023-03-22 19-15-19

File: README.md [hunk 3]
```diff
@@ -1 +1,154 @@
...
+### Key Features <a name="key-features"></a>
+
+- **Filter companies by Company Names**
+- **Display all companies by Company Name and then filter by name or symbol**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+ - [Live Demo](Coming soon)
```
Comment (amejid) [inline README.md]: - [x] As part of the project requirements, we are supposed to deploy and make the app accessible online. Let's work on it. I suggest using Netlify as it is easier and optimized for React apps. 🥸️

**Project requirements**
Screenshot from 2023-03-22 19-18-10

File: README.md [hunk 4]
```diff
@@ -1 +1,154 @@
...
+
```
Comment (amejid) [inline README.md:57]: - [x] Kindly record video documentation as suggested in the project requirements. When recording the video, please keep in mind the following requirements. 🧐️

**Project requirements**
Screenshot from 2023-03-22 19-18-37

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @mrkamin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct use of Gitflow :heavy_check_mark: 
- Great UI design :heavy_check_mark: 
- Good test coverage :heavy_check_mark: 
- Most functionalities working as expected :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/Home.js (hunk 1 above)
Comment (mrkamin) [reply src/pages/Home.js]: thanks for your good and valuable feedback.

File: src/_tests_/Home.test.js (hunk 2 above)
Comment (mrkamin) [reply src/_tests_/Home.test.js:21]: thanks for your good and valuable feedback.

File: README.md (hunk 3 above)
Comment (mrkamin) [reply README.md]: thanks for your good and valuable feedback.

File: README.md (hunk 4 above)
Comment (mrkamin) [reply README.md:57]: thanks for your good and valuable feedback.

**Comment by mrkamin** [comment]
Submit For Second Time Review.

File: src/pages/Home.js (hunk 1 above)
Comment (rloterh) [reply src/pages/Home.js]: resolved! ✔️

File: README.md (hunk 3 above)
Comment (rloterh) [reply README.md]: resolved! ✔️

File: README.md (hunk 4 above)
Comment (rloterh) [reply README.md:57]: Needs attention!

File: README.md [hunk 5]
```diff
@@ -1 +1,157 @@
...
+- **Display all companies by Company Name and then filter by name or symbol**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+ - Live Demo: [netlify](https://superb-puppy-322d9f.netlify.app/)
+
+ **Loom Video Link**
+[Description of project](https://www.loom.com/share/7d89819ba3d94f64ac37fb8c3e12494f)
```
Comment (rloterh) [inline README.md]: - [x] Good video presentation, however, it's rather too long than expected. As per the requirements for video presentation, let keep the video length to a maximum of **5 minutes**.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Mohammad, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!

## Highlights 👍
- Good job acknowledging Nelson Sakwa on Behance for the [Original design idea](https://www.behance.net/sakwadesignstudio).
Correctly deployed project!
- App works as intended!
- Good ReadMe!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 5 above)
Comment (mrkamin) [reply README.md]: Tanks for your good and valuable feedback

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @mrkamin 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Great video presentation
- Excellent communication with the reviewers

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._


@MahmoudBakr23
thanks for your good and valuable feedback.

**Pull request merged**